### DISCUSSION TITLE
More eyes needed on the docs!

### DISCUSSION THREAD

**Discussion by Griatch** [evennia/evennia#2989] [Evennia Engine Development]
We are drawing near to 1.0 release. Since this all goes with new documentation, it could be good with more eyes on things though: 

- [Installation docs](https://www.evennia.com/docs/1.0-dev/Setup/Installation.html) - we have a new installation process; you will now be able to install evennia with `pip install evennia`. Evennia 1.0rc6 is on pypi and you can test it out. **Please report any issues or quirks, especially if you are on Windows or Mac.**
- [Beginner Tutorial](https://www.evennia.com/docs/1.0-dev/Howtos/Howtos-Overview.html#beginner-tutorial) - while not finished yet (the full tutorial won't be done until after 1.0 release), the Intro and the lessons of Part1 should be. **Please help with looking it over/trying it to see if things make sense**. 
- Please report everything you find as a [documentation issue](https://github.com/evennia/evennia/issues/new/choose) (or make a PR to fix). 

I will keep working through reformatting and cleaning up various parts of the docs in the coming week. There are also some recent bugs I want to address. If you have some unreported bug or PR you want to be addressed in 1.0, this is the last shot to get them in. 

(Of course development won't stop with 1.0. We'll continue with 1.1 etc :) )

**Comment by InspectorCaracal** [comment]
The new installation docs are great, but the requirements section is incomplete for Windows. Including the python path is mentioned, but the [MS Visual C++ build tools](https://aka.ms/vs/16/release/vs_buildtools.exe) is also required. It's linked to on the troubleshooting page, but it's a _hard_ requirement so it needs to be on the main installation page with the other requirements.

(And a link to a walkthrough of how to install it would be useful, too; it is kind of a pain to figure out if you don't otherwise use Visual Studio anything)

**Reply by Griatch** [reply]
The Visual Studio build tools should not be needed. Alas, I can't clear out all the MS Visual C++ installs I have on my Win10 installation to check (a lot of stuff depends on those). 

But I spun up an old, clean VM with Win 7 (gack!) and could (after allowing the package to be installed with 3.8, since this is the latest Python supporting Win7), install and run Evennia just fine. No need for extra installs. 

My guess is that this is an upstream Twisted binary package which has not been updated to latest Python 3.11 yet. :confused: Either way I suppose we need to make the build tool installation more visible, then.

**Reply by InspectorCaracal** [reply]
Ahhh! That would explain it. I was very surprised when we ran into that issue while I was walking someone through installing, but I figured it must have not been an issue for me because I already had it installed from my last attempt.

If that's the case, then having it in the installation troubleshooting section *is* the right place for it. Although a link to something explaining exactly what to install and how would still be useful.

**Reply by Griatch** [reply]
For completeness for those not in chat - I talked to the Twisted folks and it does indeed appear like Twisted has not released binary wheels for Python 3.11 yet:

[twisted-iocpsupport]( https://pypi.org/project/twisted-iocpsupport/#files) holds the Twisted Windows C binaries. Normally when you do `pip install evennia`, Windows `pip` will automatically fetch the binaries from there. Since they are not updated for 3.11 yet though, you can only download the source files and have to compile the Python C-extensions yourself. This is what requires the MS Visual C++ compiler. 

I have now updated the 1.0-dev installation documentation. If installing the C++ compilers is daunting, the fastest fix is to just use Python 3.10 for now - it has binaries to download.

**Comment by Griatch** [comment]
As a note, I think you only need the `Windows 10 SDK`; that pulls the headers etc without needing to find the individual C++ version.

**Reply by InspectorCaracal** [reply]
Aha. This is exactly the kind of thing that we need the install instructions for, yeah. xD
